angular-workspace/projects/ni/nimble-angular/pipes/format-number-decimal.pipe.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject, LOCALE_ID, Pipe, PipeTransform } from '@angular/core';
+import { DecimalUnitFormat } from '@ni/nimble-components/dist/esm/utilities/unit-format/decimal-unit-format';
+import { passthroughUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/passthrough-unit-scale';
+import { byteUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-unit-scale';
+import { byte1024UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/byte-1024-unit-scale';
+import { voltUnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/volt-unit-scale';
+import type { UnitScale } from '@ni/nimble-components/dist/esm/utilities/unit-format/unit-scale/unit-scale';
+
+export { byteUnitScale, byte1024UnitScale, voltUnitScale };
+
+/**
+ * A pipe that transforms a decimal number into a string with optional unit, formatted like "1.2 kB"
+ */
+@Pipe({
+    name: 'formatNumberDecimal',
+    standalone: true
+})
+export class FormatNumberDecimalPipe implements PipeTransform {
+    /**
+     * @internal
+     */
+    public decimalUnitFormat?: DecimalUnitFormat;
+    private minimumFractionDigits?: number;
+    private maximumFractionDigits?: number;
+    private unitScale?: UnitScale;
+
+    public constructor(@Inject(LOCALE_ID) private readonly locale: string) {}
+
+    public transform(value: number, {
+        decimalDigits,
+        maximumDecimalDigits,
+        unitScale = passthroughUnitScale
+    }: {
+        decimalDigits?: number,
+        maximumDecimalDigits?: number,
+        unitScale?: UnitScale
+    } = {
+        unitScale: passthroughUnitScale
+    }): string {
+        if (decimalDigits !== undefined && maximumDecimalDigits !== undefined) {
```

Comment:
Could we do something to enforce this requirement at compile-time rather than throwing an exception? For example, how difficult would it be to make the `transform` function look like:

```ts
public transform(value: number, options?: {
        decimalDigits?: undefined,
        maximumDecimalDigits?: number,
        unitScale?: UnitScale
    } | {
        decimalDigits?: number,
        maximumDecimalDigits?: undefined,
        unitScale?: UnitScale
    }): string {
        const configuredUnitScale = options?.unitScale ?? passthroughUnitScale;
        const configuredDecimalDigits = options?.decimalDigits;
        const configuredMaximumDecimalDigits = options?.maximumDecimalDigits;

        /* ... */
}
```